### DISCUSSION TITLE
time_varying_media and lift test

### DISCUSSION THREAD

**Discussion by malitsadok1** [pymc-labs/pymc-marketing#1311] [Q&A] [answered]
Hi , 
I tried using `time_varying_media = True` with my MMM model  and adding a lift test with the function `add_lift_test_measurements`, but I encountered an error:  
`MissingValueError: "The value ['date'] is missing from the required ['channel', 'date']"`

I know that the required values are: `channel`, `x`, `delta_x`, `delta_y`, and `sigma`. I tried adding `date` to see if it would help, but it didn't. However, when I removed `time_varying_media = True` from the model, the error disappeared.  
any idea what’s causing this or how I can fix it?

**Comment by williambdean** [comment]
Yeah, you will require a date column to map the lift test to a date in the model in the case of time varying media. Can you set the date column to one of the dates from training data set and share error.

**Comment by malitsadok1** [comment]
Thank you for the quick response.  
I added the column `date` in the following formats:  
1. "date": [pd.to_datetime("2024-08-01")]
2. "date": ["2024-08-01"]

However, I am still getting the same error:

MissingValueError                         Traceback (most recent call last)
[<ipython-input-41-a36e40a54354>](https://localhost:8080/#) in <cell line: 1>()
----> 1 mmm.add_lift_test_measurements(df_lift_test)

3 frames
[/usr/local/lib/python3.10/dist-packages/pymc_marketing/mmm/lift_test.py](https://localhost:8080/#) in assert_is_subset(required, available)
    248     missing = required - available
    249     if missing:
--> 250         raise MissingValueError(list(missing), list(required))
    251 
    252 

MissingValueError: "The value ['date'] is missing from the required ['channel', 'date']"

**Reply by williambdean** [reply, marked as answer]
Okay. Thanks, this is a bug then. Can you please raise an issue for this

**Comment by malitsadok1** [comment]
thanks!
